File: go-algorithm/leetcode/problem1137.go
package main

// 1137. N-th Tribonacci Number
// https://leetcode.com/problems/n-th-tribonacci-number/
func tribonacci(n int) int {
	memo := make(map[int]int)
	return recur(n, memo)
}
func recur(n int, memo map[int]int) int {
	if n == 0 {
		return 0
	}
	if n == 1 || n == 2 {
		return 1
	}
	if memo[n] != 0 {
		return memo[n]
	}
	memo[n] = recur(n-3, memo) + recur(n-2, memo) + recur(n-1, memo)
	return memo[n]
}

File: java-algo/src/main/java/com/leetcode/Problem1137.java
package com.leetcode;

/**
 * 1137. N-th Tribonacci Number
 * https://leetcode.com/problems/n-th-tribonacci-number/
 */
public class Problem1137 {
    public int tribonacci(int n) {
        if (n < 2) {
            return n;
        }
        if (n == 2) {
            return 1;
        }
        int[] dp = new int[n + 1];
        dp[0] = 0;
        dp[1] = 1;
        dp[2] = 1;
        for (int i = 3; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2]+ dp[i - 3];
        }
        return dp[n];
    }
}
